sycl/include/CL/sycl/accessor.hpp

```diff
@@ -381,39 +391,49 @@ class image_accessor
         T, detail::type_list<cl_int4, cl_float4>>::type::value;
   };
 
-  //  Available only when: (accessTarget == access::target::image ||
-  //  accessTarget == access::target::host_image) && accessMode ==
-  //  access::mode::read
+  // Available only when:
+  // (accessTarget==access::target::image && accessMode==access::mode::read) ||
+  // (accessTarget==access::target::host_image &&
+  //  accessMode==access::mode::read_write)
   template <typename CoordT, int Dims = Dimensions,
             typename = detail::enable_if_t<
                 (Dims > 0) && (IsValidCoordDataT<Dims, CoordT>::value) &&
-                (IsImageAcc || IsHostImageAcc) && IsImageAccessRead>>
+                ((IsImageAcc && IsImageAccessReadOnly) ||
+                 (IsHostImageAcc && IsImageAccessAnyRead))>>
   DataT read(const CoordT &Coords) const {
     // TODO: To be implemented.
     throw cl::sycl::feature_not_supported("Read API is not implemented.");
     return;
   };
 
-  // Available only when: (accessTarget == access::target::image || accessTarget
-  // == access::target::host_image) && accessMode == access::mode::read
+  // Available only when:
+  // (accessTarget==access::target::image && accessMode==access::mode::read) ||
+  // (accessTarget==access::target::host_image &&
+  //  accessMode==access::mode::read_write)
```

Comment:
```suggestion
  //  (accessMode == access::mode::read || accessMode == access::mode::read_write))
```

Reply:
done

Reply:
NOTE: `accessMode == access::mode::read` check was not added as suggested.